### DISCUSSION TITLE
is 'multi process read, one process write' storage already in the v8 EA?

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#521] [Q&A] [answered]
Hi, per an article i read online version 8 should have:
`We have now a read-only mode so that you can have multiple processes accessing the same storage, as long as only one process makes changes and the other only access theta in read-only mode.`
is this already in 08.00.00-MS-EA2?
if a read process already read specific data that is after that changed by the write process. will this leave the data in the read process stale? how would that process know? should read process always reload data from scratch?
are there docs for this? any special API?
thanks!

**Comment by fh-ms** [comment, marked as answer]
The read-only mode is intended for debugging purposes or for replica nodes in the cluster. Theoretically, it will work at startup of the nodes, but they will get no update information, because there is no communication between the nodes. The current MicroStream version is designed that only a single process can access a storage. The new generation of MicroStream, which is in alpha right now, will be able to do that.

**Reply by dcoracle** [reply]
Do you have any details on when this new generation of Microstream will be released?  Also do you have a high level architecture on how this will be implemented?  I hope the simplicity is still there and we don't have to deal with a client/server model.  At the very least I hope it can be abstracted in the API (similar to how embedded H2 deals with auto-mixed mode)

**Reply by hg-ms** [reply]
We haven't set a release date yet and, very important, we currently don’t plan to release this version publicly.
The new architecture will be a server/client model but one of our goals to keep it’s API as simple as possible like the current API.
